core/store/src/flat/delta.rs

```diff
@@ -109,6 +109,15 @@ impl FlatStateChanges {
         Self(delta)
     }
 
+    pub fn from_raw_key_value(entries: &[(Vec<u8>, Option<Vec<u8>>)]) -> Self {
```

Comment:
Sanity check: does this handle deletions correctly by virtue of using an Option?

Reply:
Yep, this is eventually consumed in `flat_state_changes.apply_to_flat_state` function -> `store_helper::set_flat_state_value`
```
    let db_key = encode_flat_state_db_key(shard_uid, &key);
    match value {
        Some(value) => store_update
            .set_ser(DBCol::FlatState, &db_key, &value)
            .expect("Borsh should not have failed here"),
        None => store_update.delete(DBCol::FlatState, &db_key),
    }
```